### DISCUSSION TITLE
Running MATLAB live scripts in Neurodesk

### DISCUSSION THREAD

**Discussion by civier** [neurodesk/neurodesk.github.io#121] [Ideas]
I've just discovered MATLAB live scripts (similar to Python's Jupiter), and I wonder if we can run them from the compiled versions of EEGLAB, FieldTrip and SPM. I think it will be very useful, as one big limitation of the compiled code is the difficulty to know what line of code generated which output (because compiled code cannot be run line by line).

@neuro-sevas Would you like to try converting one of your EEGLAB MATLAB script into .mlx (the Live Script extension) and try to run it from the EEGLAB container?

Judy, would you like to do the same with FieldTrip and SPM? (If you don't have an SPM script at had, maybe we can ask @DavidjWhite33).

Best,
Oren

**Comment by JD-Zhu** [comment]
Hi Oren, I just gave this a try - it looks like the compiled Fieldtrip doesn't work with .mlx files though (please see below):

<img width="566" alt="Capture" src="https://user-images.githubusercontent.com/38392787/146143150-936df16f-21fa-435c-9ee7-675329818955.PNG">

**Comment by civier** [comment]
Thanks a lot for checking it @JD-Zhu At least we tried :-(

I think that the next thing to check is if Matlab Live Scripts can be compiled at all? If yes, it might be worth compiling popular neuroimaging pipelines that are provided as Live Scripts (if any exist).
Have you seen any mention of compiling Live Scripts in the MATLAB documentation?

**Reply by JD-Zhu** [reply]
I have just tried compiling a live script - it seems to work in the same way as a normal .m script. However, the compiled live script also runs in the same way (i.e. no Live Editor), so it doesn't really achieve our purpose. Not sure if there are more components that can be added to the Matlab runtime to support Live Editor.

**Comment by civier** [comment]
Thanks for investigating it Judy. I wish I knew. Definitely something to
look into in the future.
If you find online any neuroimaging group that uses Live Scripts, it's
worth asking them (I haven't seen anything so far).

**Reply by JD-Zhu** [reply]
Thanks Oren - if I come across any relevant groups I'll ask them :)
